File: LeetCode Problems/401 to 500/452. Minimum Number of Arrows to Burst Balloons/452. Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
private:
    static bool cmp(vector<int> &a, vector<int> &b) {
        return a[1] < b[1];
    }

public:
    int findMinArrowShots(vector<vector<int>> &points) {
        sort(points.begin(), points.end(), cmp);

        int ans = 0, rep = 0;
        for (int i = 0; i < points.size(); i++) {
            if (ans == 0 || points[i][0] > rep) {
                ans++;
                rep = points[i][1];
            }
        }
        return ans;
    }
};

File: LeetCode Problems/401 to 500/452. Minimum Number of Arrows to Burst Balloons/452. Minimum Number of Arrows to Burst Balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (int[] a, int[] b) -> Integer.compare(a[1], b[1]));
        int ans = 0, res = 0;
        for (int i = 0; i < points.length; i++) {
            if (ans == 0 || points[i][0] > res) {
                ans++;
                res = points[i][1];
            }
        }
        return ans;
    }
}
